### PR TITLE
casm-4232-iuf-logging

### PR DESCRIPTION
**Pull request by Srinivas-Anand-HPE**
Summary and Scope
This change is to support enhanced logging for scripts used by IUF operations. The "loaders.py" script is used by the "ims-upload-content" operation of the "deliver-product" stage of IUF. The changes introduce log levels to each statement that logs information for user.

See the following link for the feedback which prompted the changes:
https://rndwiki-pro.its.hpecorp.net/display/~john.navitsky@hpe.com/IUF+Stage+Output

Issues and Related PRs
None

Testing
Planned for test in #mug .

Tested on:

Test description:
Run the "deliver-product" stage for the "COS" product to verify.

Risks and Mitigations
None

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+## [2.5.1] - 2023-05-31
+
+### Changed
+
+- CASM-4232: Enhanced logging for loaders.py for use with IUF.
```
Comment (mharding-hpe) [inline CHANGELOG.md:14]:
```suggestion
- CASM-4232: Enhanced logging for [`loaders.py`](ims_load_artifacts/loaders.py) for use with IUF.
```

**Review by mharding-hpe — CHANGES_REQUESTED** [review]
I am confused by the nature of the changes. The changelog text and the PR description state that this is meant to enhance the logging. However, almost no new logging is added. In some cases the log message is changed in a tiny way (removing the final period from the output). In all of the other cases, the logging is changed from "info" level to "debug" level, which means it will show up LESS often in the logs (since DEBUG is a more detailed logging level than INFO).  In a couple of instances the log message is improved to include the filename (where it currently does not), but it also changes the log level from info to debug, meaning that the message itself will show up only if debug logging is enabled.

In other words, the effect of this PR will only SLIGHTLY increase the information being logged if debug logging is already enabled (in which case you will get those filenames in the places where previously you did not). Otherwise you will get LESS information (if a log level less than DEBUG is set). Is this the desired behavior? Is there a reason we want these log messages to be rarer?

The only part of this PR which makes sense to me is adding the filename to the messages where it is absent. I don't see why the info messages are changed to debug.

**Comment by sravani-sanigepalli** [comment]
> I am confused by the nature of the changes. The changelog text and the PR description state that this is meant to enhance the logging. However, almost no new logging is added. In some cases the log message is changed in a tiny way (removing the final period from the output). In all of the other cases, the logging is changed from "info" level to "debug" level, which means it will show up LESS often in the logs (since DEBUG is a more detailed logging level than INFO). In a couple of instances the log message is improved to include the filename (where it currently does not), but it also changes the log level from info to debug, meaning that the message itself will show up only if debug logging is enabled.
> 
> In other words, the effect of this PR will only SLIGHTLY increase the information being logged if debug logging is already enabled (in which case you will get those filenames in the places where previously you did not). Otherwise you will get LESS information (if a log level less than DEBUG is set). Is this the desired behavior? Is there a reason we want these log messages to be rarer?
> 
> The only part of this PR which makes sense to me is adding the filename to the messages where it is absent. I don't see why the info messages are changed to debug.


Hi @mharding-hpe. I completely get where you are coming from. So these changes are made majorly from the IUF perspective. There are so many images that IUF uses in various stages. So when an IUF command is run, the log messages comes from various places - IUF itself, NLS and from these images that IUF calls. So from the user perspective its a bit unclear of actually what is happening. User would be getting loads of messages which might not be quite information for them. That is the reason we planned to change most of the messages which are not much critical to 'DEBUG'(which would be going into the logs in IUF) and leave critical messages alone in 'INFO' which would appear on screen when IUF commands are run so it will be less cluttered. Hope this answers your queries.
